dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProjectServiceImpl.java

```diff
@@ -799,4 +809,85 @@ public Result queryAllProjectListForDependent() {
+        }
+        List<Project> projects = projectMapper.selectBatchIds(projectIds);
+
+        List<Long> projectCodes = projects.stream().map(project -> project.getCode()).collect(Collectors.toList());
+
+        workflowCountResponse.setWorkflowCounts(projectMapper.queryAllWorkflowCounts(projectCodes));
+
+        putMsg(workflowCountResponse, Status.SUCCESS);
+        return workflowCountResponse;
+    }
+
+    /**
+     * query all workflow States count
+     * @param loginUser login user
+     * @param statisticsStateRequest statisticsStateRequest
+     * @return workflow States count
+     */
+    @Override
+    public Map<String, Object> countWorkflowStates(User loginUser,
+                                                   StatisticsStateRequest statisticsStateRequest) {
+        Map<String, Object> result = new HashMap<>();
+        Set<Integer> projectIds = resourcePermissionCheckService
+                .userOwnedResourceIdsAcquisition(AuthorizationType.PROJECTS, loginUser.getId(), logger);
+        if (projectIds.isEmpty()) {
+            putMsg(result, Status.SUCCESS);
+            return result;
+        }
+        String projectName = statisticsStateRequest.getProjectName();
+        String workflowName = statisticsStateRequest.getWorkflowName();
+        Date date = new Date();
+        Date startTime = statisticsStateRequest.getStartTime() == null ? DateUtils.getMonthAgo(date)
+                : statisticsStateRequest.getStartTime();
+        Date endTime = statisticsStateRequest.getEndTime() == null ? date : statisticsStateRequest.getEndTime();
+        Integer model = Constants.QUERY_ALL_ON_SYSTEM;
+        if (!StringUtils.isBlank(projectName)) {
+            model = Constants.QUERY_ALL_ON_PROJECT;
+        }
+        if (!StringUtils.isBlank(workflowName)) {
+            model = Constants.QUERY_ALL_ON_WORKFLOW;
+        }
+        List<ExecuteStatusCount> executeStatusCounts = processInstanceMapper.countInstanceStateV2(
+                startTime, endTime, projectName, workflowName, model, projectIds);
+        TaskCountDto taskCountResult = new TaskCountDto(executeStatusCounts);
+        result.put(Constants.DATA_LIST, taskCountResult);
+        putMsg(result, Status.SUCCESS);
+        return result;
+    }
+
+    @Override
+    public Map<String, Object> countOneWorkflowStates(User loginUser, String workflowName) {
+        Map<String, Object> result = new HashMap<>();
+        Project project = projectMapper.queryByName(workflowName);
+        boolean hasProjectAndWritePerm = hasProjectAndWritePerm(loginUser, project, result);
+        if (!hasProjectAndWritePerm) {
+            return result;
+        }
+        List<ExecuteStatusCount> executeStatusCounts = processInstanceMapper.countInstanceStateV2(
+                null, null, null, workflowName, Constants.QUERY_ALL_ON_WORKFLOW, null);
+        if (executeStatusCounts != null) {
+            TaskCountDto taskCountResult = new TaskCountDto(executeStatusCounts);
```

Comment:
I think we should not use taskcout dto in workflow stats. but it is odd that the exists function `countProcessInstanceStateByProject` also use this function. do you know why they use them ?